spec/custom_matchers.rb

```diff
@@ -0,0 +1,12 @@
+RSpec::Matchers.define :belong_to do |expected|
+  match { |actual| (actual.to_a.uniq - expected.to_a.uniq).empty? }
+end
+
+RSpec::Matchers.define :contain_from do |expected|
+  match do |actual|
+    actual = actual.to_a.map { |x| x.data.to_a }.flatten.uniq
+    expected.map!(&:values) unless expected.first.is_a? Array
+    expected = expected.flatten.uniq
+    (actual - expected).empty?
+  end
+end
```

Comment:
I believe that all test cases here should be "exact". E.g. not "some data including this data", but "exactly THIS data (but ignore the order)".
More boldly, I am even not sure that "ignore the order" is a really good idea. Maybe "default sort by something, with option to resort by something else" is more reasonable?